### PR TITLE
added 0xCB Static

### PR DESCRIPTION
**Pull request by Conor-Burns**
## Description

Added the TH board 0xCB static with a default keymap and a keymap with via support enabled

The is a 40% with an OLED and an encoder -> [hardware](https://github.com/0xCB-dev/0xcb-static)
Also thank you @drashna for all the help with the OLED code!

## Types of Changes

- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Issues Fixed or Closed by This PR

*  No Issues

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/0xcb/static/rules.mk [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+MCU = atmega328p
+
+# Bootloader selection
+BOOTLOADER = USBasp
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes 	    # Mouse keys
```
Comment (fauxpark) [inline keyboards/0xcb/static/rules.mk:11]:
```suggestion
MOUSEKEY_ENABLE = yes       # Mouse keys
```

File: keyboards/0xcb/static/rules.mk [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no	        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
```
Comment (fauxpark) [inline keyboards/0xcb/static/rules.mk:21]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Conor-Burns** [comment]
Yeah, Travis keeps failing but I don't think thats my fault :D

**Review by fauxpark — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

File: keyboards/0xcb/1337/1337.c [hunk 3]
```diff
@@ -24,9 +24,11 @@ void eeconfig_init_kb(void) {
 #ifdef RGBLIGHT_ENABLE
     rgblight_enable();
     rgblight_sethsv(0, 255, 255);
-#ifdef RGBLIGHT_ANIMATIONS
+// clang-format off
+    #ifdef RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/0xcb/1337/1337.c:28]:
```suggestion
#    ifdef RGBLIGHT_ANIMATIONS
```

File: keyboards/0xcb/1337/1337.c [hunk 4]
```diff
@@ -24,9 +24,11 @@ void eeconfig_init_kb(void) {
...
     rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL + 2);
-#endif
+    #endif
+// clang-format on
```
Comment (drashna) [inline keyboards/0xcb/1337/1337.c:31]:
```suggestion
#    endif
```

File: keyboards/0xcb/1337/config.h [hunk 5]
```diff
@@ -17,6 +17,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 #include "config_common.h"
+// clang-format off
```
Comment (drashna) [inline keyboards/0xcb/1337/config.h:20]:
```suggestion
```

File: keyboards/0xcb/1337/config.h [hunk 6]
```diff
@@ -60,6 +61,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
   #define RGBLIGHT_EFFECT_RAINBOW_MOOD
 #endif
+// clang-format on
```
Comment (drashna) [inline keyboards/0xcb/1337/config.h:64]:
```suggestion

```

File: keyboards/0xcb/static/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    _______, KC_MUTE, KC_VOLU, KC_VOLD, KC_MPRV, KC_MNXT, KC_MSTP, KC_INS,  KC_HOME, KC_DEL,  KC_END,  _______,
+    _______, _______, KC_BRIU, KC_BRIU, _______, _______, _______, KC_PGUP, KC_UP,   KC_PGDN,          _______,
+    _______,          _______, _______, _______, _______, _______, _______, KC_LEFT, KC_DOWN, KC_RGHT, _______,
+    _______, _______, _______, _______,          _______,                   _______, _______, _______, _______
+),
+};
+// clang-format on
+
+/*layer switcher */
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/0xcb/static/keymaps/via/keymap.c:58]: Also, via has keycodes to handles this, actually. 
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

Specifically `FN_MO13` and `FN_MO23`

File: keyboards/0xcb/static/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+    _______, KC_MUTE, KC_VOLU, KC_VOLD, KC_MPRV, KC_MNXT, KC_MSTP, KC_INS,  KC_HOME, KC_DEL,  KC_END,  _______,
+    _______, _______, KC_BRIU, KC_BRIU, _______, _______, _______, KC_PGUP, KC_UP,   KC_PGDN,          _______,
+    _______,          _______, _______, _______, _______, _______, _______, KC_LEFT, KC_DOWN, KC_RGHT, _______,
+    _______, _______, _______, _______,          _______,                   _______, _______, _______, _______
+),
+};
+// clang-format on
+
+/*layer switcher */
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/0xcb/static/keymaps/default/keymap.c:58]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/0xcb/static/readme.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+To go to bootloader press ESC while plugging in or hold the RESET switch, then hold the BOOT switch, release RESET, release BOOT.
+The board should now appear in lsusb (or device manager).
+
+Make example for this keyboard (after setting up your build environment):
+
+    make 0xcb/static:default
+
+Flashing example for this keyboard:
+
+    make 0xcb/1337:static:flash
```
Comment (fauxpark) [inline keyboards/0xcb/static/readme.md:25]:
```suggestion
    make 0xcb/static:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by Erovia — APPROVED** [review]

**Comment by Conor-Burns** [comment]
@Erovia thanks, mate!

**Pull request merged**